oss_src/flexible_type/flexible_type_converter.hpp

```diff
@@ -13,621 +13,619 @@
+    
+    return conditional_test<is_sequence_container<T>::value && is_std_pair<pair_type>::value,
+                            is_flexible_type_convertible, pair_type>::value;
+  }
 
-    flex_vec f = val.to<flex_vec>();
-    std::vector<T> ret;
-    ret.resize(f.size());
-    for (size_t i = 0;i < f.size(); ++i) ret[i] = f[i];
-    return ret;
+  template <typename T, typename U, template <typename...> class C>
+  static void get(C<std::pair<T, U> >& dest, const flexible_type& src) {
+    if(src.get_type() == flex_type_enum::DICT) {
+      const flex_dict& fd = src.get<flex_dict>();
+      dest.resize(fd.size());
+      for(size_t i = 0; i < fd.size(); ++i) {
+        convert_from_flexible_type(dest[i].first, fd[i].first);
+        convert_from_flexible_type(dest[i].second, fd[i].second);
+      }
+    } else if(src.get_type() == flex_type_enum::LIST) {
+      const flex_list& fl = src.get<flex_list>();
+      dest.resize(fl.size());
+      for(size_t i = 0; i < fl.size(); ++i) {
+        convert_from_flexible_type(dest[i], fl[i]);
+      }
+    } else {
+      throw_type_conversion_error(src, "flex_dict or flex_list of 2-element list/vectors");
+    }
   }
-  flexible_type set(const std::vector<T>& val) {
-    flex_vec f;
-    f.resize(val.size());
-    for (size_t i = 0;i < val.size(); ++i) f[i] = val[i];
-    return f;
+
+  template <typename T, typename U, template <typename...> class C>
+  static void set(flexible_type& dest, const C<std::pair<T, U> >& src) {
+   flex_dict d(src.size());
+    for(size_t i = 0; i < src.size();++i) {
+      d[i] = std::make_pair(convert_to_flexible_type(src[i].first), convert_to_flexible_type(src[i].second));
+    }
+    dest = std::move(d); 
   }
 };
 
+////////////////////////////////////////////////////////////////////////////////
 
-/**
- * Case 5: std::vector<T> where T is is of any type convertible to flexible_type.
- * Must exclude the types already handled by case 1, and the numeric types
- * handled by case 3. Converts to flex_list.
+/** Any pairs of numeric values. 
  */
-template <typename T>
-struct flexible_type_converter<std::vector<T>, 
-    typename std::enable_if<is_flexible_type_convertible<T>::value &&
-                            !is_flexible_type_member<std::vector<T>>::value &&
-                            !std::is_arithmetic<T>::value>::type> {
-  static constexpr bool value = true;
-  std::vector<T> get(const flexible_type& val) {
-
-    if (val.get_type() != flex_type_enum::LIST) {
-      std::string errormsg = std::string("Expecting a list, But we got a ") 
-          + flex_type_enum_to_name(val.get_type());
-      throw(errormsg);
-    }
+template <> struct ft_converter<7> {
+  
+  template <typename T> static constexpr bool matches() {
+    typedef typename first_nested_type<T>::type  U1; 
+    typedef typename second_nested_type<T>::type U2; 
+    
+    return (is_std_pair<T>::value
+            && std::is_arithmetic<U1>::value 
+            && std::is_arithmetic<U1>::value); 
```

Comment:
What is this case?
should this be is_arithmetic<U2>?  Why is this case distinct from the case below?